misc/tracing-extensions/modules/ballerina-jaeger-extension/src/main/java/org/ballerinalang/observe/trace/extension/jaeger/OpenTracingExtension.java

```diff
@@ -49,43 +51,59 @@
 @JavaSPIService("org.ballerinalang.util.tracer.OpenTracer")
 public class OpenTracingExtension implements OpenTracer {
 
+    private ConfigRegistry configRegistry;
+    private String hostname;
+    private int port;
+    private String samplerType;
+    private Number samplerParam;
+    private int reporterFlushInterval;
+    private int reporterBufferSize;
+
     private static final PrintStream console = System.out;
     private static final PrintStream consoleError = System.err;
-    private Map<String, String> configProperties;
 
     @Override
-    public void init(Map<String, String> configProperties) {
-        console.println("ballerina: started publishing tracers to Jaeger on "
-                + configProperties.getOrDefault(REPORTER_HOST_NAME_CONFIG, DEFAULT_REPORTER_HOSTNAME) + ":" +
-                getValidIntegerConfig(configProperties.get(REPORTER_PORT_CONFIG),
-                        DEFAULT_REPORTER_PORT, REPORTER_PORT_CONFIG));
-        this.configProperties = configProperties;
+    public void init() throws InvalidConfigurationException {
+        configRegistry = ConfigRegistry.getInstance();
+
+        try {
+            port = Integer.parseInt(
+                    configRegistry.getConfigOrDefault(REPORTER_PORT_CONFIG, String.valueOf(DEFAULT_REPORTER_PORT)));
+            hostname = configRegistry.getConfigOrDefault(REPORTER_HOST_NAME_CONFIG, DEFAULT_REPORTER_HOSTNAME);
+
+            samplerType = configRegistry.getConfigOrDefault(SAMPLER_TYPE_CONFIG, DEFAULT_SAMPLER_TYPE);
+            if (!(samplerType.equals(ConstSampler.TYPE) || samplerType.equals(RateLimitingSampler.TYPE)
+                    || samplerType.equals(ProbabilisticSampler.TYPE))) {
+                samplerType = DEFAULT_SAMPLER_TYPE;
+                consoleError.println(
+                        "ballerina: Jaeger configuration: \"sampler type\" invalid. Defaulted to const sampling");
+            }
+
+            samplerParam = Float.valueOf(
+                    configRegistry.getConfigOrDefault(SAMPLER_PARAM_CONFIG, String.valueOf(DEFAULT_SAMPLER_PARAM)));
+            reporterFlushInterval = Integer.parseInt(configRegistry.getConfigOrDefault(
+                    REPORTER_FLUSH_INTERVAL_MS_CONFIG, String.valueOf(DEFAULT_REPORTER_FLUSH_INTERVAL)));
+            reporterBufferSize = Integer.parseInt(configRegistry.getConfigOrDefault
```

Comment:
Why do we need Integer.parseInt() now? ConfigRegistry should have an API to getOrDefault for int as well.. @pubudu91 WDYT?